diff --git a/dist/index.cjs b/dist/index.cjs
index 915f00bf81f45196ae56a4fa85bc65b23bd70f71..cd29ae7c421f638e1ee8a3f78de8ff581a8f1352 100644
--- a/dist/index.cjs
+++ b/dist/index.cjs
@@ -159,7 +159,7 @@ function getComponentDetailsTemplate(component, options, isComment = false) {
     options == null ? void 0 : options.hideMethodDocs,
     (_e = options == null ? void 0 : options.labels) == null ? void 0 : _e.methods
   );
-  let description = getComponentDescription(component, options == null ? void 0 : options.descriptionSrc) + "\n---\n" + events + methods + slots + cssProps + parts;
+  let description = getComponentDescription(component, options == null ? void 0 : options.descriptionSrc) + "\n\n---\n\n" + events + methods + slots + cssProps + parts;
   if (isComment) {
     description = description.split("\n").map((x) => ` * ${x}`).join("\n");
   }
diff --git a/dist/index.js b/dist/index.js
index 9f8ecef9566bb8de3c7ec7e6e413af724183f8e1..c9d26b49fc0ee1d082722ce1a6ad4041a7cc1feb 100644
--- a/dist/index.js
+++ b/dist/index.js
@@ -120,7 +120,7 @@ function getComponentDetailsTemplate(component, options, isComment = false) {
     options == null ? void 0 : options.hideMethodDocs,
     (_e = options == null ? void 0 : options.labels) == null ? void 0 : _e.methods
   );
-  let description = getComponentDescription(component, options == null ? void 0 : options.descriptionSrc) + "\n---\n" + events + methods + slots + cssProps + parts;
+  let description = getComponentDescription(component, options == null ? void 0 : options.descriptionSrc) + "\n\n---\n\n" + events + methods + slots + cssProps + parts;
   if (isComment) {
     description = description.split("\n").map((x) => ` * ${x}`).join("\n");
   }
