diff --git a/dist/chunk-C44SADVH.js b/dist/chunk-C44SADVH.js
index a5dc5f5a2542f9531e876b35a81b1ddc2712d2a0..c463aee341d12844452e3b29994438ddf7c64621 100644
--- a/dist/chunk-C44SADVH.js
+++ b/dist/chunk-C44SADVH.js
@@ -2,13 +2,15 @@
 import { debug, getElementLocatorSelectors } from "@vitest/browser/utils";
 import { render as litRender } from "lit";
 var containers = /* @__PURE__ */ new Set();
-function render(template, {
+async function render(template, {
   baseElement = document.body,
   container = baseElement.appendChild(document.createElement("div")),
   ...options
 } = {}) {
   containers.add(container);
   litRender(template, container, options);
+  // Let Lit components finish rendering
+  await new Promise((resolve) => requestAnimationFrame(() => resolve()));
   return {
     container,
     baseElement,
@@ -17,8 +19,9 @@ function render(template, {
       containers.delete(container);
       container.remove();
     },
-    rerender(newTemplate) {
+    async rerender(newTemplate) {
       litRender(newTemplate, container, options);
+      await new Promise((resolve) => requestAnimationFrame(() => resolve()));
     },
     asFragment() {
       return document.createRange().createContextualFragment(container.innerHTML);
diff --git a/dist/pure.d.ts b/dist/pure.d.ts
index ebd3ac24afe257f3c51153ff23d88decb5a5182d..79df7dcb0e4c70ed7702e39ed50513e39d6eca01 100644
--- a/dist/pure.d.ts
+++ b/dist/pure.d.ts
@@ -11,10 +11,10 @@ interface RenderResult extends LocatorSelectors {
     baseElement: HTMLElement;
     debug: typeof debug;
     unmount: () => void;
-    rerender: (template: unknown) => void;
+    rerender: (template: unknown) => Promise<void>;
     asFragment: () => DocumentFragment;
 }
-declare function render(template: unknown, { baseElement, container, ...options }?: ComponentRenderOptions): RenderResult;
+declare function render(template: unknown, { baseElement, container, ...options }?: ComponentRenderOptions): Promise<RenderResult>;
 declare function cleanup(): void;
 
 export { type ComponentRenderOptions, type RenderResult, cleanup, render };
